supportedBackends/google/pipelines/v2alpha1/src/main/scala/cromwell/backend/google/pipelines/v2alpha1/GenomicsFactory.scala

```diff
@@ -47,7 +58,75 @@ case class GenomicsFactory(applicationName: String, authMode: GoogleAuthMode, en
       genomics.projects().operations().get(job.jobId).buildHttpRequest()
     }
 
-    override def runRequest(createPipelineParameters: CreatePipelineParameters, jobLogger: JobLogger) = {
+    override def runRequest(createPipelineParameters: CreatePipelineParameters, jobLogger: JobLogger): HttpRequest = {
+
+      def projectMetadataRequest(vpcConfig: VirtualPrivateCloudConfiguration): IO[HttpRequest] = {
+        val workflowOptions = createPipelineParameters.jobDescriptor.workflowDescriptor.workflowOptions
+
+        val googleCredentialOption = vpcConfig.auth.apiClientGoogleCredential((key: String) => workflowOptions.get(key).get)
+
+        googleCredentialOption match {
+          case None => IO.raiseError(new RuntimeException(s"Unable to find Google Credential for auth `${vpcConfig.auth.name}`."))
+          case Some(googleCredential) => IO {
+            val auth = googleCredential.createScoped(ResourceManagerAuthScopes)
+
+            val cloudResourceManagerBuilder = new CloudResourceManager
+              .Builder(GoogleAuthMode.httpTransport, GoogleAuthMode.jsonFactory, auth)
+              .build()
+
+            val project = cloudResourceManagerBuilder.projects().get(createPipelineParameters.projectId)
+
+            project.buildHttpRequest()
+          }
+        }
+      }
+
+
+      def projectMetadataResponseToLabels(httpResponse: HttpResponse): IO[ProjectLabels] = {
+        IO.fromEither(decode[ProjectLabels](httpResponse.parseAsString())).handleErrorWith {
+          e => IO.raiseError(new RuntimeException(s"Failed to parse labels from project metadata response from Google Cloud Resource Manager API. " +
+            s"${ExceptionUtils.getMessage(e)}", e))
+        }
+      }
+
+
+      def networkFromLabels(vpcConfig: VirtualPrivateCloudConfiguration, projectLabels: ProjectLabels): Network = {
+        val networkLabelOption = projectLabels.labels.find(l => l._1.equals(vpcConfig.name))
+
+        networkLabelOption match {
+          case Some(networkLabel) =>
+            new Network()
+              .setUsePrivateAddress(createPipelineParameters.runtimeAttributes.noAddress)
+              .setName(VirtualPrivateCloudNetworkPath.format(createPipelineParameters.projectId, networkLabel._2))
+          case None =>
+            jobLogger.debug(s"Project `${createPipelineParameters.projectId}` does not have high security network configured. " +
+              s"Project metadata does not have network label key `${vpcConfig.name}`. Falling back to running the job on default network.")
```

Comment:
I am not sure if we want to spam our logs with this when the project doesn't have the `network-label-key`. Once this change is configured in Cromwell for Terra, initially only few projects will be using high security networks, and hence majority of the jobs will fallback to using the default network. But I wasn't sure if this is helpful for debugging purposes or not, so I added it. If we think this is not needed and will simply spam our logs, I can remove it.

Reply:
Would it make sense to have it as `INFO` rather than `DEBUG`?

Reply:
Hmmm... personally I don't think so. 

I think it's a normal and expected situation for a project not to have this label set, so I personally would probably just remove the log line altogether.